server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/SenderAuthIdentifyVerificationRcptHook.java

```diff
@@ -38,11 +39,13 @@
 public class SenderAuthIdentifyVerificationRcptHook extends AbstractSenderAuthIdentifyVerificationRcptHook {
     private final DomainList domains;
     private final UsersRepository users;
+    private final CanSendFrom canSendFrom;
 
     @Inject
-    public SenderAuthIdentifyVerificationRcptHook(DomainList domains, UsersRepository users) {
+    public SenderAuthIdentifyVerificationRcptHook(DomainList domains, UsersRepository users, CanSendFrom canSendFrom) {
         this.domains = domains;
         this.users = users;
+        this.canSendFrom = canSendFrom;
```

Comment:
allowedSender?

Reply:
I prefer `canSendFrom` because it's the instance of the object which allow to check if a user can send from a mail. And not the result of the the validation for a specific user/address.

Reply:
I do not see the difference.

For me it's not following a logical reasoning "an user *can send* with this *from*" only if "this *sender* is *allowed*".

`canSendFrom` looks like a fixed anterior result.

Reply:
Sender sounds ambiguous : is it the connected user or the address one want to use of a `From:` header in a mail?